File: app/src/main/java/wangdaye/com/geometricweather/db/generator/ChineseCityEntityGenerator.java
package wangdaye.com.geometricweather.db.generator;

import java.util.ArrayList;
import java.util.List;

import wangdaye.com.geometricweather.basic.model.ChineseCity;
import wangdaye.com.geometricweather.db.entity.ChineseCityEntity;

public class ChineseCityEntityGenerator {

    public static ChineseCityEntity generate(ChineseCity city) {
        ChineseCityEntity entity = new ChineseCityEntity();
        entity.cityId = city.getCityId();
        entity.province = city.getProvince();
        entity.city = city.getCity();
        entity.district = city.getDistrict();
        entity.latitude = city.getLatitude();
        entity.longitude = city.getLongitude();
        return entity;
    }

    public static List<ChineseCityEntity> generateEntityList(List<ChineseCity> cityList) {
        List<ChineseCityEntity> entityList = new ArrayList<>(cityList.size());
        for (ChineseCity city : cityList) {
            entityList.add(generate(city));
        }
        return entityList;
    }

    public static ChineseCity generate(ChineseCityEntity entity) {
        return new ChineseCity(
                entity.cityId,
                entity.province, entity.city, entity.district,
                entity.latitude, entity.longitude
        );
    }

    public static List<ChineseCity> generateModuleList(List<ChineseCityEntity> entityList) {
        List<ChineseCity> cityList = new ArrayList<>(entityList.size());
        for (ChineseCityEntity entity : entityList) {
            cityList.add(generate(entity));
        }
        return cityList;
    }
}

File: app/src/main/java/org/breezyweather/db/generators/ChineseCityEntityGenerator.kt
package org.breezyweather.db.generators

import org.breezyweather.common.basic.models.ChineseCity
import org.breezyweather.db.entities.ChineseCityEntity

object ChineseCityEntityGenerator {
    fun generate(city: ChineseCity): ChineseCityEntity {
        return ChineseCityEntity(
            cityId = city.cityId,
            province = city.province,
            city = city.city,
            district = city.district,
            latitude = city.latitude,
            longitude = city.longitude
        )
    }

    fun generateEntityList(cityList: List<ChineseCity>): List<ChineseCityEntity> {
        val entityList: MutableList<ChineseCityEntity> = ArrayList(cityList.size)
        for (city in cityList) {
            entityList.add(generate(city))
        }
        return entityList
    }

    fun generate(entity: ChineseCityEntity): ChineseCity {
        return ChineseCity(
            entity.cityId,
            entity.province,
            entity.city,
            entity.district,
            entity.latitude,
            entity.longitude
        )
    }

    fun generateModuleList(entityList: List<ChineseCityEntity>): List<ChineseCity> {
        val cityList: MutableList<ChineseCity> = ArrayList(entityList.size)
        for (entity in entityList) {
            cityList.add(generate(entity))
        }
        return cityList
    }
}
